### ISSUE TITLE
Legend filter (color) usage question

### ISSUE THREAD

**Issue by bboure** [ant-design/ant-design-charts#2288]
Hi, and thanks for contributing to this library. 

I recently started using ant-design-charts for one of my projects. I am using v2

I struggle to find documentation/information about achieving legend filtering.

I can filter the data when I click on a "color" legend item.

One problem is that the filter does not persist after the data changes. 

Also, I would like to persist the filters across page reloads (i.e. I want to keep them in the local storage).

in v1.4, I found [this page](https://ant-design-charts.antgroup.com/en/examples/component/legend/#legend-focus) where I ca programmatically enable/disable a filter. 

I believe this might what I would need in v2, but I cannot to find the equivalent. 

![Filters not persisted](https://github.com/ant-design/ant-design-charts/assets/7089997/55c3f4c2-4337-4f24-a599-2184ff771667)

Expected behavior:
- The filters are persisted after a data refresh / new data
- I should be able to specify which are the "default" filters

Desired API suggestion
```ts
<Line
  data={data}
  legend={{
    color: {
      position: 'bottom',
      selected: {
        'Latency (Average)': true,
        'Latency (p10)': false,
        'Latency (p90)': false,
      },
      // or, alternatively
      selected: ['Latency (Average)']
    },
  }}
  // ... other props
/>
```

Thank you!

**Comment by lxfu1** [comment]
https://codesandbox.io/p/sandbox/sleepy-hamilton-cfxyqw

**Comment by bboure** [comment]
Thank you @lxfu1 

I tried that already and it does not work well for several reasons: 

As I mentioned, when the data changes (refresh), the graph filters are reset. I tried using several things like refiltering after `afterchangedata` (that event never triggers) and other events, but it would cause other issues. It also looks like a hack and is not ideal.

Also, using `setTimeout` causes a delay, which is not very user-friendly, and also does not guarantee anything. e.g. if loading the data takes more than the timeout time, it does not work

I wish there was a better solution. It could be an idea for an improvement.

Thank you anyway.
